dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -111,25 +97,72 @@ const getSizes = (
 	) {
 		// Showcase main media images (which includes numbered list articles) appear
 		// larger than in body showcase images so we use a different set of image sizes
-		return `(min-width: ${breakpoints.wide}px) 1020px, (min-width: ${breakpoints.leftCol}px) 940px, (min-width: ${breakpoints.tablet}px) 700px, (min-width: ${breakpoints.phablet}px) 660px, 100vw`;
+		if (breakpoint >= breakpoints.wide) return 1020;
+		if (breakpoint >= breakpoints.leftCol) return 940;
+		if (breakpoint >= breakpoints.tablet) return 700;
```

Comment:
magic numbers could these be named consts ?

Reply:
I thought about this function a lot, on it's own it kind of acts as a source-of-truth for what sizes images should be based on their role, breakpoint, etc, which means the numbers mostly bare similarities with each other only by coincidence. I don't know if moving them into consts or another format would bring much benefit, but perhaps the comments on this function could be updated to help describe what the numbers actually mean?

Reply:
One thing you can do if you want a bit of type safety is extracting all the possible options, [like so][]:

```ts
const imageWidths = [
    1020,
    940,
    700,
    680,
    620,
    320,
    300,
    /* … */
] as const;

type Breakpoints = typeof breakpoints[keyof typeof breakpoints]
type Pixel = Breakpoints | typeof imageWidths[number];
```

[like so]: https://www.typescriptlang.org/play?ssl=20&ssc=55&pln=8&pc=1#code/MYewdgzgLgBARgJwKYEMDWAHEBLMUIwC8MA3gFAyUwC2Ic2ANkgFwwBsADBwDQVVQo4TKKwDsAFh59KAEyQQ0UEBlYBOTryowA7tjmsAjAGYumqgHoAVDEBkBDEvmyAXxgoCoSFADcZD9BjY1CgA5kgA6npQABYExADa0jAGHABMUlqqkmaUoqaJbAAc6VRsadkwRmWJJsWUVrb2jgC6ru7g0D5kUACeGEgwAELI6Fi4+EQwPX0gAGbww5g4eBBxaEjds5O9SJuIqItjEE1d2zAACtgAHkgME0P7o8swAD5b03OBIeGRMXFgAK7UOBIBBNLxAA